File: frontend/src/app/services/basket.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { JsonResponse } from "../models/jsonResponse";
import { environment } from "src/environments/environment";
import { Size } from "../models/size";

@Injectable({
    providedIn: 'root'
})
export class BasketService {
      
    private url = "basket";
    constructor(private http: HttpClient) { }

    addBasketItem(productId: number, size: Size): Observable<JsonResponse> {
        const params = new HttpParams()
        .set('productId', productId.toString())
        .set('size', size.toString());
        return this.http.post<JsonResponse>(`${environment.baseApiUrl}/${this.url}/addBasketItem`, {}, {params});
    }

    removeBasketItem(basketItemId: number): Observable<JsonResponse> {
        return this.http.delete<JsonResponse>(`${environment.baseApiUrl}/${this.url}/removeBasketItem/${basketItemId}`, {});
    }

    decreaseQuantityFromBasketItem(basketItemId: number): Observable<JsonResponse> {
        return this.http.put<JsonResponse>(`${environment.baseApiUrl}/${this.url}/decreaseQuantityFromBasketItem/${basketItemId}`, {});
    }


}

File: backend/backend/src/main/java/rs/ac/uns/ftn/service/BasketService.java
package rs.ac.uns.ftn.service;

import rs.ac.uns.ftn.model.BasketItem;
import rs.ac.uns.ftn.model.Size;

public interface BasketService {

    BasketItem addBasketItem(Long productId, Size size, String loggedUserEmail);

    BasketItem removeBasketItem(Long basketItemId, String loggedUserEmail);

    BasketItem decreaseQuantityFromBasketItem(Long basketItemId, String loggedUserEmail);
}
